### DISCUSSION TITLE
Aligning an object within another object when rotated

### DISCUSSION THREAD

**Discussion by ShuffledInk** [fabricjs/fabric.js#8835] [Q&A]
I asked this question on SO but I feel like this is probably a better place for it.

[Here is my SO question](https://stackoverflow.com/questions/75988809/fabricjs-aligning-an-object-to-the-left-of-another-object)

**Comment by jpomykala** [comment]
Use bounding rectangle to check collisions http://fabricjs.com/bounding-rectangle

**Reply by ShuffledInk** [reply]
I did take a look at that prior to asking. Just struggling on how to set the bounding rect left of the object to the left of the parent object

**Reply by jpomykala** [reply]
My apologies, I didn’t get that at the beginning. 

You can try to get a bounding box of the rectangle, and use the `left` value to calculate a proper `left` value for the rectangle. 
I guess that in your case it might be just `canvasArea.left + Math.abs(boundingRect.left)`

**Reply by ShuffledInk** [reply]
I managed to figure it out with some calculations. Thanks so much for your response! You led me down the right path.

**Reply by jpomykala** [reply]
Calculations are always the right path while working with coordinate systems. 😄
